src/tools/cephfs/JournalTool.cc

```diff
@@ -866,8 +866,11 @@ int JournalTool::recover_dentries(
         // Compose: Dentry format is dnfirst, [I|L], InodeStore(bare=true)
         bufferlist dentry_bl;
         encode(fb.dnfirst, dentry_bl);
-        encode('I', dentry_bl);
+        encode('i', dentry_bl);
+        ENCODE_START(2, 1, dentry_bl);
+        encode(fb.alternate_name, dentry_bl);
```

Comment:
Is this really the encode order, where Inode has alternate_name at the front and Link has it at the back?

Reply:
This is correct.